### PR TITLE
FIREFLY-1431:  Refresh IRSA Catalog Search (UI Conversion)

### PR DESCRIPTION
**Pull request by loitly**
Ticket: https://jira.ipac.caltech.edu/browse/FIREFLY-1431

- Rewrite and rename CatalogSelectViewPanel.jsx to IrsaCatalogSearch.jsx.  Remove old code files.
- Add OptionListField.jsx; a wrapper for JoyUI's List component.
- Modify CatalogSearchMethodType.jsx to remove jumpy behavior during search type selection.
- Minor cleanup done to FormPanel.jsx to allow more customization.
- Minor layout changes to CatalogConstraintsPanel.jsx

Tests:  IRSA Catalog search tab/panel
https://fireflydev.ipac.caltech.edu/firefly-1431-irsa-cat-search/firefly/
https://firefly-1431-irsa-cat-search.irsakudev.ipac.caltech.edu/irsaviewer/

- [ ] During loading, page render footprint(skeleton); removing flickering seen in the old UI.
- [ ] Search button is rendered on the bottom and not scrolled off when window is smaller.
- [ ] Page expands/shrinks correctly when window resizes
- [ ] Table selection field has the new JoyUI look
- [ ] Search Method is moved to the top; removing jumpiness seen in the old UI.
- [ ] Functionally, it should work the same as before.

### CODE REVIEW

**Comment by lrebull** [comment]
It looks good, and it seems to work as I expect it to work. 
It would be nice if I could grab and drag the boundary between the top two boxes (the list of IRSA tables and the 'search method' box), but hardly required!

**Comment by robyww** [comment]
Just quick UI note: in other places we have the values as warning color and the label as normal
Screenshot 2024-03-11 at 12 32 26 PM

**Review by dencoded — APPROVED** [review]

File: src/firefly/js/visualize/ui/IrsaCatalogSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+        </Sheet>
+    );
+}
+
+function catTableRenderer(opt) {
+
+    const keyVal = (k, v) => (
+        <Stack direction='row' spacing={1}>
+            <Typography level='body-sm' color='warning'>{k}</Typography>
+            <Typography level='title-sm'>{v}</Typography>
```
Comment (robyww) [inline src/firefly/js/visualize/ui/IrsaCatalogSearch.jsx:135]: Mentioned earlier: I think you should flip these.

**Review by robyww — APPROVED** [review]
One UI comment but otherwise looks good.

**Pull request merged**